Move towards native audio

masonschafercodes opened the pull request:
Here is what I have done so far:
- converted the existing sound files in `web/public/sounds` to the correct format
- created the `awc` file with the correct samples and sample rate for each sound
- created the `dat` file with all the sound types and soundset
- updated the `fxmanifest` to reflect the newly added files for the streamed audio

I figure there is a good chunk of work around the Lua code and some on the UI side to move towards using this instead of audio through NUI. Just opening this in case anyone was to work at it as well and to track the progress of it.

masonschafercodes committed:
feat(audio): awc and dat files created

converted the ogg files that were originally used to the correct format and created the awc and dat files to use in the code.

masonschafercodes committed:
chore(audio): update soundsets

thelindat commented:
> opening this in case anyone was to work at it as well and to track the progress of it.

Mark as draft?

masonschafercodes committed:
chore(audio): rename wav files for consistency

masonschafercodes committed:
feat(audio/server): load new sound names for UI

masonschafercodes committed:
feat(client): play native audio instead of NUI

masonschafercodes committed:
chore(UI): remove old playSound handler

not needed anymore

masonschafercodes committed:
Merge branch 'overextended:main' into qwade-native-audio

FjamZoo commented:
Reused my very hard written code and not even added me as a co-contributor I feel betrayed

masonschafercodes commented:
> Reused my very hard written code and not even added me as a co-contributor I feel betrayed

sorry Zoo ♥️

LukeWasTakenn reviewed (commented):
This would break all custom sounds people may have put into their sounds folder, perhaps a fallback to NUI playing sounds or automatic conversion would make sense?

thelindat commented:
> perhaps a fallback to NUI playing sounds or automatic conversion would make sense

~~Ideally scrap NUI entirely but I'd like to see a tool to convert files (somebody was already working on it).~~

Ehbw commented on audio/dlc_oxdoorlock/oxdoorlock.awc.xml:
```diff
@@ -0,0 +1,107 @@
+    </Item>
+    <Item>
+     <Type>format</Type>
+     <Codec>PCM</Codec>
```
`ADPCM` should be used over `PCM` for its smaller sizes with no noticeable lost in quality

masonschafercodes replied:
we only figured this out today you rat, but you right I will update it 🤪

FjamZoo replied:
> we only figured this out today you rat, but you right I will update it 🤪

Stop with the excuses and just do what you are told you minion

masonschafercodes committed:
use ADPCM instead of PCM format

smaller resulting file size with no noticeable quality loss

thelindat commented:
> perhaps a fallback to NUI playing sounds or automatic conversion would make sense?

Let's keep the NUI there as fallback to keep things simple; I don't mind merging after that.

masonschafercodes commented:
> Let's keep the NUI there as fallback to keep things simple; I don't mind merging after that.

Perhaps introducing a new convar could be a solution to this, something like `setr doorlock:nativeaudio [true|false]`? Let me know if this isn't what you want to do.

thelindat commented:
> Perhaps introducing a new convar could be a solution to this, something like `setr doorlock:nativeaudio [true|false]`? Let me know if this isn't what you want to do.

I'd like to include native audio as the go-to, but that also means supporting people in converting their ogg files to native audio; and breaking the sounds until then.

Sure there's now a guide on it and people are working on a tool to make it quicker, but I'd rather not deal with complaints about "newer versions breaking audio".

masonschafercodes committed:
chore(audio): bring back support for NUI audio

this seems to be a good fix to me. Just added a convar and determine whether to use native audio or NUI audio based on that. Default will be native audio (can be switched around), but NUI audio can be used by setting 'doorlock:nativeaudio' to false.

masonschafercodes commented:
I can also PR docs to mention the new convar and native audio support if need be.
